### ISSUE TITLE
Feature: Option to save collection of connected and configured modules as custom templates accessible across all apps under an account

### ISSUE THREAD

**Issue by adhilmunna** [wit-ai/wit#2589]
**Saving**
Step 1 - highlight and select a bunch of configured modules
Step 2 - Add as a new template
Step 3 - Associate intents, entities and utterances associated with all modules involved

**Inserting template**
Select from a list of saved templates and paste it onto the canvas
in background - Create and train, associated intents, entities and utterances

**Why 1** - This will reduce development time by a lot. Like a lot. Very useful when you are building similar bots for different clients. Can re-use majority of bot parts thus saving a lot of time. 

**Why 2** - Will help in scaling up bot building team. Need not train and make builders super experts if there is an option to quickly select and insert logical and functional blocks of preconfigured modules.

**Why uploading a downloaded copy of another bot is not an alternative to this?**
Presence of app specific utterances, intents and entities. Will have to delete irrelevant ones. Adds to time.
Fishing out and deleting unwanted portions from imported app leads to lot of unintentional errors and trouble shooting becomes  tedious.

**Bonus**
Option to share these templates (kind of a community contributed templates repository) will help other community members as well. Especially if the template option comes with a wiki like description option. Can help noobs easily adopt and deploy wit in projects.

**Comment by adhilmunna** [comment]
@adhilmunna @harisvsulaiman

**Comment by patapizza** [comment]
Thanks for the suggestion. We've been thinking about this for a while - hopefully we'll be able to get this out of the gate someday.

**Comment by adhilmunna** [comment]
@patapizza can we expect it soon?

**Comment by patapizza** [comment]
No, this is going to take time. The sharing model requires to also support sharing intents/entities/traits, which is non-trivial.

**Comment by adhilmunna** [comment]
@patapizza makes sense. It is indeed a complex module.
